optuna/samplers/_gp/sampler.py

```diff
@@ -289,7 +333,17 @@ def sample_independent(
         param_name: str,
         param_distribution: BaseDistribution,
     ) -> Any:
-        self._raise_error_if_multi_objective(study)
+        if len(study.directions) > 1:
+            warnings.warn(
+                f"Multi-objective optimization by {self.__class__.__name__} is an experimental "
+                "feature. The interface can change in the future."
+            )
+            if self._constraints_func is not None:
+                raise ValueError(
+                    f"{self.__class__.__name__} has not supported multi-objective optimization "
+                    "with constraints."
+                )
```

Comment:
This error should be raised more earlier. How about raising in the `sample_relative`?